app/models/user.rb

```diff
@@ -8,12 +8,14 @@ class User < ApplicationRecord
 
   CST = 'Central Time (US & Canada)'
 
-  def current_week
-    if weeks.last && Time.now.in_time_zone(CST) < weeks.last.end
-      weeks.last
+  def find_week(week)
+    week_cst_time_zone = (Time.now + week).in_time_zone(CST)
+
+    if week_exists = weeks.find_by(end: week_cst_time_zone.end_of_week)
+      week_exists
     else
-      weeks.create(start: Time.now.in_time_zone(CST).beginning_of_week,
-                   end: Time.now.in_time_zone(CST).end_of_week)
+      weeks.create(start: week_cst_time_zone.beginning_of_week,
+                   end:   week_cst_time_zone.end_of_week)
     end
   end
```

Comment:
I like how you want to dry up the code with this `find_week` method, but I think its a little brittle.
the method `find_week(week)` looks like its getting a week object, but you are actually passing in a duration object (`0.week` or `1.week`) from what I can tell. 

Instead of passing that, I think we should just pass a boolean. (This will almost likely change in the long future, but it will be a solid method for our purposes right now)

```ruby
def find_week(next_week = false) #by default it will return current week
  week_end_time = Time.now.in_time_zone(CST).end_of_week
  week_end_time = week_end_time + 1.week if next_week

  week =  weeks.find_or_create_by(end: week_end_time) #may want to use first or initialize instead
end
```